ublox-bluetooth.go

```diff
@@ -515,15 +515,9 @@ func (ub *UbloxBluetooth) GetDeviceCommsStats(mac string) (*SensorCommsStatitics
 
 // GetSensorCommsStatsDelta returns the delta of the initial and current statistics
 func (ub *UbloxBluetooth) GetSensorCommsStatsDelta(mac string, initial *SensorCommsStatitics) (*SensorCommsStatitics, error) {
-	stats, ok := ub.CommsStats[mac]
-	if !ok {
-		return nil, fmt.Errorf("No stats for device %v on dongle %v", mac, ub.GetDeviceIndex())
-	}
+	finishStats := ub.GetSerialPortStats()
 	return &SensorCommsStatitics{
-		TotalBytesTxed:    stats.TotalBytesTxed - initial.TotalBytesTxed,
-		TotalBytesRxed:    stats.TotalBytesRxed - initial.TotalBytesRxed,
-		TotalConnections:  stats.TotalConnections - initial.TotalConnections,
-		ConnectionsFailed: stats.ConnectionsFailed - initial.ConnectionsFailed,
-		TimeCommunicating: stats.TimeCommunicating - initial.TimeCommunicating,
+		TotalBytesTxed: finishStats.TxBytes - initial.TotalBytesTxed,
+		TotalBytesRxed: finishStats.RxBytes - initial.TotalBytesRxed,
 	}, nil
```

Comment:
This will break the stats display in the UI. As I alluded to earlier, if you want the stats for the connect call, a cleaner implementation would be to connect up the onDisconnect handler and get that to handle them which seems the sensible place to do it. Problem in returning the stats as you have done, is you lose the context as to which connection the stats refer to.